File: src/algebra1_compound_inequalities_easy.py
"""
algebra1 - compound_inequalities (easy)
Generated: 2026-02-22T03:51:43.857103
"""

from problem_utils import *

def generate():
    # For compound inequalities at 1000-1300 ELO, we want simple problems
    # Types: "and" conjunctions or "or" disjunctions with single-step inequalities
    
    problem_type = choice(['and_basic', 'or_basic', 'and_solve', 'or_solve'])
    
    if problem_type == 'and_basic':
        # Simple "and" with x already isolated: a < x < b
        a_val = randint(-10, 5)
        b_val = randint(a_val + 2, a_val + 10)
        
        question = f"Solve: ${a_val} < x < {b_val}$"
        answer = fmt_interval(a_val, b_val, True, True)
        difficulty = (1000, 1100)
        solution = steps(
            f"This is already in standard form.",
            f"$x$ is between ${a_val}$ and ${b_val}$ (not including endpoints).",
            f"Interval notation: $({a_val}, {b_val})$"
        )
    
    elif problem_type == 'or_basic':
        # Simple "or" with x already isolated: x < a or x > b
        a_val = randint(-5, 5)
        b_val = randint(a_val + 3, a_val + 10)
        
        question = f"Solve: $x < {a_val}$ or $x > {b_val}$"
        answer = f"(-inf,{a_val}) U ({b_val},inf)"
        difficulty = (1050, 1150)
        solution = steps(
            f"This is a disjunction (or statement).",
            f"$x$ is less than ${a_val}$ OR greater than ${b_val}$.",
            f"Interval notation: $(-\\infty, {a_val}) \\cup ({b_val}, \\infty)$"
        )
    
    elif problem_type == 'and_solve':
        # Two-step "and": a < x + b < c, need to subtract b
        b_val = nonzero(-5, 5)
        a_val = randint(-10, 5)
        c_val = randint(a_val + 3, a_val + 12)
        
        left_bound = a_val - b_val
        right_bound = c_val - b_val
        
        question = f"Solve: ${a_val} < x + {b_val} < {c_val}$"
        answer = fmt_interval(left_bound, right_bound, True, True)
        difficulty = (1150, 1300)
        solution = steps(
            f"Subtract ${b_val}$ from all parts:",
            f"${a_val} - {b_val} < x + {b_val} - {b_val} < {c_val} - {b_val}$",
            f"${left_bound} < x < {right_bound}$",
            f"Interval notation: $({left_bound}, {right_bound})$"
        )
    
    else:  # or_solve
        # Simple "or" requiring one-step solving: x + a < b or x + a > c
        a_val = nonzero(-5, 5)
        b_val = randint(-8, 3)
        c_val = randint(b_val + 4, b_val + 12)
        
        left_bound = b_val - a_val
        right_bound = c_val - a_val
        
        question = f"Solve: $x + {a_val} < {b_val}$ or $x + {a_val} > {c_val}$"
        answer = f"(-inf,{left_bound}) U ({right_bound},inf)"
        difficulty = (1200, 1300)
        solution = steps(
            f"Solve each inequality separately by subtracting ${a_val}$:",
            f"$x + {a_val} < {b_val}$ gives $x < {left_bound}$",
            f"$x + {a_val} > {c_val}$ gives $x > {right_bound}$",
            f"Interval notation: $(-\\infty, {left_bound}) \\cup ({right_bound}, \\infty)$"
        )
    
    return problem(
        question=question,
        answer=answer,
        difficulty=difficulty,
        topic="algebra1/compound_inequalities",
        solution=solution,
        answer_type="interval"
    )

emit(generate())

File: src/algebra1_compound_inequalities_easy.jl
# algebra1 - compound_inequalities (easy)
# Generated: 2026-03-08T19:54:50.791121

include(joinpath(@__DIR__, "..", "julia", "src", "ProblemUtils.jl"))
using .ProblemUtils

@script x begin
    set_topic!("algebra1/compound_inequalities")
    
    problem_type = choice([:and_basic, :or_basic, :and_solve, :or_solve])
    
    if problem_type == :and_basic
        # Simple compound inequality: a < x < b
        a = randint(-15, 10)
        b = randint(a + 3, a + 20)
        
        ans = fmt_interval(a, b, false, false)
        
        problem(
            question="Solve the compound inequality: \$$(a) < x < $(b)\$",
            answer=ans,
            difficulty=(700, 850),
            answer_type="interval",
            solution=steps(
                "The compound inequality \$$(a) < x < $(b)\$ means x is between $(a) and $(b)",
                "Both bounds are strict (not included)",
                sol("Answer", ans)
            ),
            time=45
        )
        
    elseif problem_type == :or_basic
        # Simple OR inequality: x < a or x > b
        a = randint(-10, 5)
        b = randint(a + 5, a + 20)
        
        ans = "(-∞, $(a)) ∪ ($(b), ∞)"
        
        problem(
            question="Solve the compound inequality: \$x < $(a)\$ or \$x > $(b)\$",
            answer=ans,
            answer_type="union",
            difficulty=(750, 900),
            solution=steps(
                "This is an OR compound inequality",
                "x is either less than $(a) OR greater than $(b)",
                sol("Answer", ans)
            ),
            time=60
        )
        
    elseif problem_type == :and_solve
        # Solve compound inequality with algebra: a < mx + n < b
        m = nonzero(-5, 5)
        n = randint(-12, 12)
        
        left_bound = randint(-20, 10)
        right_bound = randint(left_bound + 5, left_bound + 25)
        
        # Solve: left_bound < mx + n < right_bound
        # (left_bound - n)/m < x < (right_bound - n)/m
        x_left = (left_bound - n) // m
        x_right = (right_bound - n) // m
        
        if m > 0
            ans = fmt_interval(x_left, x_right, false, false)
            ineq_str = "$(left_bound) < $(tex(m*x + n)) < $(right_bound)"
        else
            # When dividing by negative, flip inequality
            ans = fmt_interval(x_right, x_left, false, false)
            ineq_str = "$(left_bound) < $(tex(m*x + n)) < $(right_bound)"
        end
        
        problem(
            question="Solve the compound inequality: \$$(ineq_str)\$",
            answer=ans,
            answer_type="interval",
            difficulty=(900, 1100),
            solution=steps(
                sol("Given", ineq_str),
                "Subtract $(n) from all parts: \$$(left_bound - n) < $(tex(m*x)) < $(right_bound - n)\$",
                "Divide all parts by $(m)" * (m < 0 ? " (flip inequalities since dividing by negative)" : ""),
                sol("Answer", ans)
            ),
            time=90
        )
        
    else  # or_solve
        # OR inequality: ax + b < c or dx + e > f
        a = nonzero(-6, 6)
        b = randint(-15, 15)
        c = randint(-20, 20)
        
        d = nonzero(-6, 6)
        e = randint(-15, 15)
        f = randint(-20, 20)
        
        # Solve ax + b < c => x < (c - b)/a or x > (c - b)/a
        bound1 = (c - b) // a
        # Solve dx + e > f => x > (f - e)/d or x < (f - e)/d
        bound2 = (f - e) // d
        
        if a > 0
            left_part = "(-∞, $(bound1))"
        else
            left_part = "($(bound1), ∞)"
        end
        
        if d > 0
            right_part = "($(bound2), ∞)"
        else
            right_part = "(-∞, $(bound2))"
        end
        
        ans = "$(left_part) ∪ $(right_part)"
        
        problem(
            question="Solve: \$$(tex(a*x + b)) < $(c)\$ or \$$(tex(d*x + e)) > $(f)\$",
            answer=ans,
            answer_type="union",
            difficulty=(1000, 1200),
            solution=steps(
                "Solve each inequality separately",
                "First: \$$(tex(a*x + b)) < $(c)\$ gives \$x $(a > 0 ? "<" : ">") $(bound1)\$",
                "Second: \$$(tex(d*x + e)) > $(f)\$ gives \$x $(d > 0 ? ">" : "<") $(bound2)\$",
                "Combine with union (OR)",
                sol("Answer", ans)
            ),
            time=120
        )
    end
end